### ISSUE TITLE
btrbk.timer runs too soon after a fresh boot

### ISSUE THREAD

**Issue by wolfgangrupprecht** [digint/btrbk#486]
As is, btrbk.timer runs so early during the boot process that the hostname hasn't even been assigned yet.  This also happens long before vital services are started.

To demonstrate, shut the backup server down before midnight and boot after midnight.  Then run:

journalctl --since 0:0:0

Notice that btrbk runs in the midst of everything slowly being started up.  On my fedora-36 system the hostname was still the temporary boot placeholder name of "fedora" instead of its actual hostname.  The network wasn't up and dns wouldn't have been working either.

I'm not sure what one is supposed to wait for boot to finish under systemd, but it seems that other services simply wait a minute or two after boot before firing their timer.   I've hacked my timer file to include

btrbk.timer:

[Timer]
...
OnActiveSec=2min

There is probably a more legit systemd way of doing this, but I have no idea what that would be.

**Comment by digint** [comment]
I'm afraid I'm no expert on systemd (I have strong opinions on it and therefore don't use it whenever possible).

That said, I think you would need to add something like this in btrbk.service:
```
After=network.target
```

I'm not sure if it is wise to add something like this as default, as in many setups btrbk runs without network (and what happens if there is no `network.target`? Is there always a `network.target`?)

Please report back if that works, in the meantime I will add a "help wanted" flag on this issue.

**Comment by wolfgangrupprecht** [comment]
I'm afraid I never agreed with the direction systemd was taking things so I resisted leaning about it until now.   Hopefully someone that understands this in their gut will have a better feel for how to solve this dilemma.   

network.target seemed to soon anyway since there was potentially still lots of setup that could happen after it.   There is another very late stage target called default.target which is mentioned in "man systemd.special".   That's what I'm using locally in both the .timer and .service files.   I may not need it in both.

[Unit]
Description=...
After=default.target

**Comment by PsyBlade42** [comment]
Another aspect of this is that on my system the timer runs before the filesystems it tries to snapshot are actually mounted. Failing for obvious reasons.

I probably won't be looking into this since I want to keep my cronjobs for the time being.
But afaik After=foo only does something if foo is going to be started and ignored otherwise.

**Comment by sidju** [comment]
One pair for each use-case would make sense and make it easy to pick the one that best suits one's needs. As long as they are clearly named and documented, of course.

(I like this especially since it makes it easy for users with specific needs to modify the services so the offline one runs `btrbk snapshot` and the network one runs `btrbk resume`.)

**Comment by amessina** [comment]
I tried the `After=network-online.target` first but that didn't always mean that name resolution was totally ready.

So instead, I use a systemd snippet to workaround this on my desktop machine which isn't usually awake at midnight: `/etc/systemd/system/btrbk.timer.d/timer.conf`

```
[Timer]
OnBootSec=5m
Persistent=no
```

**Comment by Strit** [comment]
Yeah, the network-online.target didn't work for me either, but the OnBootSec did, so I'm gonna see if a "source" implemenation of it works and if it does, I'll do a PR for that instead.
